### PR TITLE
python3Packages.mayavi: update from Python 2 to Python 3 (and switch from wxPython to pyqt5)

### PR DESCRIPTION
**Pull request by bcdarwin**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bcdarwin** [comment]
@GrahamcOfBorg build python37Packages.mayavi python38Packages.mayavi

**Comment by knedlsepp** [comment]
Cool you could make this work. Will try that out ASAP.

**Comment by knedlsepp** [comment]
Built and tried and example visualization from: https://docs.enthought.com/mayavi/mayavi/auto/mlab_helper_functions.html#plot3d
Works fine (on current stable). :+1:

**Review by knedlsepp — APPROVED** [review]
Thanks for making this work with Python 3. Looks good to me.

File: pkgs/development/python-modules/mayavi/default.nix [hunk 1]
```diff
@@ -1,37 +1,43 @@
-{ stdenv, fetchPypi, buildPythonPackage
-, wxPython, pygments, numpy, vtk, traitsui, envisage, apptools
-, nose, mock
-, isPy3k
-}:
+{ lib, pythonPackages, wrapQtAppsHook }:
```
Comment (FRidh) [inline pkgs/development/python-modules/mayavi/default.nix:1]: Do *not* pass in `pythonPackages` in `python-modules`; always have parameters for the individual packages.

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/mayavi/default.nix (hunk 1 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/mayavi/default.nix]: Sorry, this is how the other Python packages using `pkgs.libsForQt5.callPackage` are currently structured.  Fixed.

Comment (knedlsepp) [reply pkgs/development/python-modules/mayavi/default.nix]: @FRidh: is there a particular reason why this is best practice?

Comment (FRidh) [reply pkgs/development/python-modules/mayavi/default.nix]: it's good practice to be explicit about inputs.

**Pull request merged**